01-CORE-SYSTEMS/02-QUALITY_ASSURANCE/bug_catchers/test_log_analyzer_security.py

```
@@ -0,0 +1,487 @@
+                cursor.execute("SELECT 1")
+                self.assertEqual(cursor.fetchone()[0], 1)
+        
+        # Should reuse connections from pool
+        self.assertEqual(len(self.db_manager._connections), 2)
+    
+    def test_connection_error_handling(self):
+        """Test connection error handling."""
+        # Test with invalid database path
+        invalid_db_manager = DatabaseManager("/invalid/path/db.db")
+        
+        with self.assertRaises(Exception):
```

Ruff (0.12.2)

Do not assert blind exception: `Exception`

(B017)